addon/components/u-edit/shared-modules/drag-and-drop.js

```diff
@@ -0,0 +1,88 @@
+import Uploader from '@upfluence/ember-upf-utils/uploader';
+import Configuration from '@upfluence/ember-upf-utils/configuration';
+
+export default class ModuleBuilder {
+  constructor(session, toast) {
+    Object.assign(this, { session, toast });
+  }
+
+  build(editor, element) {
+    return new Module(editor, element, this.session.data.authenticated.access_token, this.toast);
+  }
+}
+
+export class Module {
+  constructor(editor, element, token, toast) {
+    Object.assign(this, { editor, element, token, toast });
+
+    this.editor.registerModule(this);
+    this.editor.enableDragAndDrop();
+
+    this.uploaderHeaders = {
+      Scope: Configuration.scope[0]
+    };
+  }
+
+  _uploaderBuilder() {
+    let uploader = Uploader.create({
+      ajaxSettings: {
+        dataType: 'json',
+        headers: {
+          ...this.uploaderHeaders,
+          Authorization: `Bearer ${this.token}`
+        }
+      },
+      url: Configuration.uploaderUrl,
+      allowedExtensions: 'jpg,jpeg,png,gif',
+      maxSize: null
+    });
+
+    uploader
+      .on('didValidationError', (error) => {
```

Comment:
Do you mind changing that in your other PR while you are at it ?
It's not proper English 😬 
`onValidationError`, `hasValidationError` or just a plain `ValidationError` 🤷

Reply:
No I don't change that in my other PR

Reply:
I was asking if you *could* change it 🙂 
Anyway, come to think of it, It'll probably break stuff elsewhere, so forget about it !